### PR TITLE
DX12 Integration

### PR DESCRIPTION
**Pull request by ttanatb**
## Feature/Issue
DirectX 12

## Implementation/Solution
 - Added a rendering pipeline

## Screenshot
sturggle

## Tests
 - [x] Have you reviewed your own code?

### CODE REVIEW

File: Light Vox Engine/FrameResource.cpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
+#include "FrameResource.h"
+#include "Camera.h"
+
+using namespace DirectX;
+
+FrameResource::FrameResource(ID3D12Device * device, ID3D12PipelineState * pso, ID3D12DescriptorHeap * dsvHeap, ID3D12DescriptorHeap * cbvSrvHeap, D3D12_VIEWPORT * viewport, UINT frameResourceIndex)
```
Comment (arb7887) [inline Light Vox Engine/FrameResource.cpp]: I'd break this line into multiple lines, its pretty long.

File: Light Vox Engine/FrameResource.cpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+{
+	// Reset the command allocators and lists for the main thread.
+	for (int i = 0; i < LV_COMMAND_LIST_COUNT; i++)
+	{
+		ThrowIfFailed(commandAllocators[i]->Reset());
+		ThrowIfFailed(commandLists[i]->Reset(commandAllocators[i].Get(), pso.Get()));
+	}
+
+	// Reset the worker command allocators and lists.
+	for (int i = 0; i < LV_NUM_CONTEXTS; i++)
```
Comment (arb7887) [inline Light Vox Engine/FrameResource.cpp]: would it be best to change these 2 to something like
`for(auto i = 0; i < LV_COMMAND_LIST_COUNT; ++i)` ?

File: Light Vox Engine/stdafx.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+
+#define LV_FRAME_COUNT 2		//how many frames for swap chain
+#define LV_NUM_CONTEXTS 1		//for multi-threading
+
+//indices to the per-frame command list
+#define LV_COMMAND_LIST_COUNT 3	 
+#define LV_COMMAND_LIST_PRE 0
+#define LV_COMMAND_LIST_MID 1
+#define LV_COMMAND_LIST_POST 2
```
Comment (arb7887) [inline Light Vox Engine/stdafx.h]: You should put a newline at the end of each file so git doesn't get mad. Also, Pre-Processor definitions shouldn't be in the pre-compiled header; you should probably move to a separate file.

**Comment by ttanatb** [comment]
@engine-buddies/graphics-pod ready for re-review

**Review by arb7887 — APPROVED** [review]

**Pull request merged**